cpp/src/arrow/gpu/cuda_context.cc

```diff
@@ -343,5 +343,12 @@ void* CudaContext::handle() const { return impl_->context_handle(); }
 
 int CudaContext::device_number() const { return impl_->device().device_num; }
 
+Status CudaContext::GetDeviceAddress(uintptr_t addr, uintptr_t* devaddr) {
```

Comment:
It would be nice to have at least a minimal test for this API in the C++ tests.

Reply:
done